include/mfmg/dealii/amge_host.templates.hpp

```diff
@@ -26,48 +26,20 @@
 
 #include <EpetraExt_MatrixMatrix.h>
 
-// APPROACH1: use UMFPACK to invert the matrix
-// APPROACH2: use CG to invert
-// APPROACH3: do not need invert, use ARPACK regular mode instead of
-//            shift-and-invert mode
-
-#define APPROACH1
+#define MATRIX_FREE 0
 
 namespace mfmg
 {
 
-#if defined(APPROACH2)
-template <typename MatrixType>
-struct WrapInverse
+struct Identity
```

Comment:
I would add a constructor with size to store. So it's going to be an identity of specific size. And then in `vmult` I would check that the sizes of the vectors match to this size.

Reply:
Won't fix.  I want to use the default constructor.